esphome/__main__.py

```diff
@@ -287,17 +290,31 @@ def upload_program(config, args, host):
 
         return 1  # Unknown target platform
 
-    from esphome import espota2
-
     if CONF_OTA not in config:
         raise EsphomeError(
             "Cannot upload Over the Air as the config does not include the ota: "
             "component"
         )
 
+    from esphome import espota2
+
     ota_conf = config[CONF_OTA]
     remote_port = ota_conf[CONF_PORT]
     password = ota_conf.get(CONF_PASSWORD, "")
+
+    mdns_disabled = (
+        CONF_MDNS in config
+        and CONF_DISABLED in config[CONF_MDNS]
+        and config[CONF_MDNS][CONF_DISABLED] is True
+    )
```

Comment:
Please test and confirm, but `mdns` should be auto loaded if `wifi` or `ethernet` is in the config and `CONF_DISABLED` defaults to `False` so this structure should always be in the config. 
If `wifi` or `ethernet` are not there then this would fail above on `CONF_OTA not in config` so should be safe.

```suggestion
    mdns_disabled = config[CONF_MDNS][CONF_DISABLED]
```

Reply:
you are correct, I simplify-ed the code.

and also found a other but with the mqtt client id with haveing 2 dashboards on the same MQTT server. (which is fixed not too)
had my changes running in kubernetes and the testing on my computer.